### PR TITLE
[IMP] udes_stock: Block deletition of pickings outside draft state

### PR DESCRIPTION
**Pull request by LoKKeR96**
Story: x2764

Signed-off-by: Lorenzo Cucurachi <lorenzo.cucurachi@unipart.com>

### CODE REVIEW

File: addons/udes_stock/models/stock_picking.py [hunk 1]
```diff
@@ -994,3 +994,17 @@ def button_validate(self):
         res = super().button_validate()
         user_validating.unassign_pickings_from_users()
         return res
+
+    def unlink(self):
+        """
+        Override unlink to ensure that if disable_state_check context isn't set and there are
+        transfers in state different from draft to raise  error with appropriate message.
+        """
+        if not self._context.get("disable_state_check"):
```
Comment (PeterC10) [inline addons/udes_stock/models/stock_picking.py:1003]: Suggestion (feel free to resolve): `bypass_state_check` as we aren't disabling anything, just bypassing it on certain occasions.

File: addons/udes_stock/tests/test_stock_picking.py [hunk 2]
```diff
@@ -1882,3 +1882,79 @@ def test_priority_one_remains_on_cancelled_picking(self):
...
+        """
+        Test that a picking can be deleted in draft state
+        """
+        pick = self.create_picking(
+            picking_type=self.picking_type_pick,
+            products_info=[{"product": self.fig, "uom_qty": 10}],
+            location_id=self.test_stock_location_02.id,
+            confirm=False,
+            assign=False,
+            priority="0",
```
Comment (PeterC10) [inline addons/udes_stock/tests/test_stock_picking.py:1896]: Q: Does priority matter?

File: addons/udes_stock/models/stock_picking.py (hunk 1 above)
Comment (armandcela) [reply addons/udes_stock/models/stock_picking.py:1003]: I think you are right, replacing the name. Didnt pay attention on naming

File: addons/udes_stock/tests/test_stock_picking.py (hunk 2 above)
Comment (armandcela) [reply addons/udes_stock/tests/test_stock_picking.py:1896]: Dont think it matters, just an extra variable as priority is not a required field. But doesnt harm as well :)

**Review by PeterC10 — APPROVED** [review]

**Review by ws-mps — APPROVED** [review]

**Pull request merged**